### PR TITLE
added sequence labeling example

### PR DESCRIPTION
**Pull request by kahne**
added sequence labeling example

### CODE REVIEW

**Comment by sammiie** [comment]
To address the "added sequence labeling example" issue #360, you can create a sequence labeling example using Python and a machine learning framework like PyTorch or TensorFlow. Here, I'll provide a step-by-step guide with code snippets to help you add this example to your project.

Let's assume you want to implement a Named Entity Recognition (NER) task as a sequence labeling example.

mport Required Libraries:
Start by importing the necessary libraries, including PyTorch or TensorFlow and any relevant NLP packages.
import torch
import torch.nn as nn
import torch.optim as optim
from torch.utils.data import DataLoader, Dataset
from transformers import BertTokenizer, BertForTokenClassification

Load and Preprocess Data:
Load your sequence labeling dataset and preprocess it. For NER, you might have data in the CoNLL format (token, label) per line.
# Load and preprocess data (example)
with open('ner_data.txt', 'r') as file:
    lines = file.read().splitlines()

# Process lines into token-label pairs
data = []
for line in lines:
    tokens, labels = line.split('\t')
    data.append((tokens.split(), labels.split()))

Define a Custom Dataset:
Create a custom dataset class to load and process your data.
class SequenceLabelingDataset(Dataset):
    def __init__(self, data, tokenizer, label_map):
        self.data = data
        self.tokenizer = tokenizer
        self.label_map = label_map

    def __len__(self):
        return len(self.data)

    def __getitem__(self, idx):
        tokens, labels = self.data[idx]
        input_ids = self.tokenizer.convert_tokens_to_ids(tokens)
        label_ids = [self.label_map[label] for label in labels]
        return {
            'input_ids': input_ids,
            'label_ids': label_ids,
        }

Define Your Model:
You can use a pre-trained model like BERT for token classification.
tokenizer = BertTokenizer.from_pretrained('bert-base-uncased')
model = BertForTokenClassification.from_pretrained('bert-base-uncased', num_labels=num_labels)

Define Training Loop:
Implement the training loop to train your sequence labeling model.
# Define your optimizer and loss function
optimizer = optim.AdamW(model.parameters(), lr=2e-5)
loss_fn = nn.CrossEntropyLoss()

# Create DataLoader for your dataset
dataset = SequenceLabelingDataset(data, tokenizer, label_map)
dataloader = DataLoader(dataset, batch_size=batch_size, shuffle=True)

# Training loop
for epoch in range(num_epochs):
    for batch in dataloader:
        input_ids = batch['input_ids'].to(device)
        label_ids = batch['label_ids'].to(device)

        outputs = model(input_ids)
        logits = outputs.logits

        loss = loss_fn(logits.view(-1, num_labels), label_ids.view(-1))
        loss.backward()
        optimizer.step()
        optimizer.zero_grad()

Evaluation and Testing:
Implement code to evaluate your trained model on a validation dataset and test it on new sequences.

Fine-Tuning and Hyperparameters:
Fine-tune your model and adjust hyperparameters based on your specific sequence labeling task.

Documentation:
Add necessary comments and documentation to your code to make it clear and understandable.

This code serves as a starting point for adding a sequence labeling example to your project. Be sure to adapt and extend it based on your dataset and task requirements.
